src/Makefile

```diff
@@ -37,6 +37,13 @@ NODEPS:=clean distclean
 # Default settings
 STD=-pedantic -DSERVER_STATIC=''
 
+OLD_ENGINE_NAME=redis
```

Comment:
let's hold off on this change for now. 

this change serves two purposes:

1. to control the final executable name - this is now solved by the symlink
2. to control the log ident - we have agreed to keep the first release compatible with redis so this needs to be redis anyways

For the longer term, I think @madolson's recommendation of coarse compat levels (strict/loose/none) makes more sense to me at this point. I can see fine-grained compat knobs like this easily get out of control. Instead, we might want to have the SERVER_NAME definition controlled by the compat level setting. let's continue our discussion on this compat topic but decouple it from this pr.

btw I also think we need to drop the compat layers at some point but with a sufficiently long ramp down time (think years).